src/Appwrite/Platform/Tasks/ScheduleFunctions.php

```diff
@@ -68,8 +68,11 @@ protected function enqueueResources(Group $pools, Database $dbForConsole): void
             \go(function () use ($delay, $scheduleKeys, $pools) {
                 \sleep($delay); // in seconds
 
-                $queue = $pools->get('queue')->pop();
-                $connection = $queue->getResource();
+                $pool = $pools['pools-queue-queue']['pool'];
+                $connection = $pool->get();
+                $this->connections->add($connection, $pool);
```

Comment:
and what is this->connections? Whats it used for?

Reply:
check [this](https://github.com/appwrite/appwrite/pull/7975#discussion_r1745904413)